smarts/core/chassis.py

```diff
@@ -59,9 +59,13 @@
 def _query_bullet_contact_points(bullet_client, bullet_id, link_index):
     contact_objects = set()
 
-    min_, max_ = bullet_client.getAABB(bullet_id, link_index)
     # `getContactPoints` does not pick up collisions well so we cast a fast box check on the physics
+    min_, max_ = bullet_client.getAABB(bullet_id, link_index)
+    # note that getAABB returns a box around the link_index link only,
+    # which means it's offset from the ground (min_ has a positive z)
+    # if link_index=0 (the chassis link) is used.
     overlapping_objects = bullet_client.getOverlappingObjects(min_, max_)
+    # the pairs returned by getOverlappingObjects() appear to be in the form (body_id, link_idx)
     if overlapping_objects is not None:
         contact_objects = set(oo for oo, _ in overlapping_objects if oo != bullet_id)
```

Comment:
Is this still a problem? I think you were looking at doing the following:

```suggestion
        contact_objects = set(oo for _, oo in overlapping_objects if oo != bullet_id)
```

Reply:
No, it turned out it wasn't a bug, I was just misunderstanding the thing that I wrote in the comment I added.  (That's why I just added the comment instead.)